app/models/event.rb

```diff
@@ -504,6 +504,39 @@ def self.subj_id_check(options = {})
     end
   end
 
+  def self.modify_nested_objects(options = {})
+    size = (options[:size] || 1000).to_i
+    cursor = [options[:from_id], options[:until_id]]
+
+    response = Event.query(nil,  page: { size: 1, cursor: [] })
+    Rails.logger.warn "[modify_nested_objects] #{response.results.total} events for source datacite-crossref."
+
+    # walk through results using cursor
+    if response.results.total.positive?
+      while response.results.results.length.positive?
+        response = Event.query(nil, page: { size: size, cursor: cursor })
+        break unless response.results.results.length.positive?
+
+        Rails.logger.warn "[modify_nested_objects] modify_nested_objects #{response.results.results.length}  events starting with _id #{response.results.to_a.first[:_id]}."
+        cursor = response.results.to_a.last[:sort]
+        Rails.logger.warn "[modify_nested_objects] Cursor: #{cursor} "
+
+        ids = response.results.results.map(&:uuid).uniq
```

Comment:
Why are you not looping through the ids and start a job for each one of them instead of having an extra job for the list of ids?